src/s_print.c

```diff
@@ -14,6 +14,42 @@
 t_printhook sys_printhook = NULL;
 int sys_printtostderr;
 
+#ifdef _WIN32
+
+    /* NB: Unlike vsnprintf(), _vsnprintf() does *not* null-terminate
+    the output if the resulting string is too large to fit into the buffer.
+    Also, it just returns -1 instead of the required number of bytes.
+    In this case, we null-terminate the buffer and return 'size' to mimick
+    the behavior of vsnprintf().
+    Strictly speaking, the UCRT in Windows 10 actually contains a standard-
+    conforming vsnprintf() function that is not just an alias for _vsnprintf().
+    However, MinGW traditionally links against the old msvcrt.dll runtime library.
+    Recent versions of MinGW seem to have their own (standard-conformating)
+    implementation of vsnprintf(), but to ensure portability we rather use our
+    own implementation for all Windows builds. */
+int pd_vsnprintf(char *buf, size_t size, const char *format, va_list argptr)
+{
+    int ret = _vsnprintf(buf, size, format, argptr);
+    if (ret < 0)
+    {
+        buf[size - 1] = '\0';
+        return size;
+    }
+    else return ret;
+}
+
+int pd_snprintf(char *buf, size_t size, const char *format, ...)
+{
+    int ret;
+    va_list ap;
+    va_start(ap, format);
+    ret = pd_vsnprintf(buf, size, format, ap);
+    va_end(ap);
+    return ret;
+}
+
+#endif
+
```

Comment:
given my comment of having the same API for Windows and non-Windows, we would also need (thin) implementations of `pd_vsnprintf()` and `pd_snprintf()` on systems that do support `snprintf()`.

even if we typically would use `snprintf()` directly (by means of the `#define` in the header)